### PR TITLE
Timers to display as min:sec

### PR DESCRIPTION
**Pull request by kev306**
Added a function to convert ms to min:sec. Utilised function in ready prompt/game messages to display timeouts as min:sec

UI change for Timeout formats
image

### CODE REVIEW

File: src/util/timeConverterToMinSec.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export function timeConverterToMinSec(time: string) : string {
```
Comment (vck3000) [inline src/util/timeConverterToMinSec.ts:1]: Have the input this function just be in millis please.

Comment (kev306) [reply src/util/timeConverterToMinSec.ts]: Updated in subsequent commit!

File: src/util/timeConverterToMinSec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export function timeConverterToMinSec(milliseconds: number) : string {
+
+    let minutes: number = Math.floor(milliseconds/ (1000 * 60) );
+    let seconds: number = Math.floor((milliseconds / 1000) % 60);
+    let formattedSeconds: string;
+
+    if (seconds < 10) {
+        formattedSeconds = "0" + seconds;
```
Comment (vck3000) [inline src/util/timeConverterToMinSec.ts:8]: You probably need a clause like this for mins as well. Add a test first to cause this to fail and then fix it up here.

File: src/util/timeConverterToMinSec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    let seconds: number = Math.floor((milliseconds / 1000) % 60);
+    let formattedSeconds: string;
+
+    if (seconds < 10) {
+        formattedSeconds = "0" + seconds;
+    } else {
+        formattedSeconds = seconds.toString();
+    }
+
+    return minutes + ":" + formattedSeconds;
```
Comment (vck3000) [inline src/util/timeConverterToMinSec.ts:13]: Prefer backtick substitution over string concatenation please.

Comment (vck3000) [reply src/util/timeConverterToMinSec.ts:13]: I think it’s called template strings in JS

File: src/util/timeConverterToMinSec.ts (hunk 2 above)
Comment (vck3000) [reply src/util/timeConverterToMinSec.ts:8]: Careful with types. Seconds here is a number which you’re adding onto a string.

File: src/util/timeConverterToMinSec.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export function timeConverterToMinSec(milliseconds: number) : string {
```
Comment (vck3000) [inline src/util/timeConverterToMinSec.ts:1]: Nit: let’s just call it millis instead of the full name.

Comment (vck3000) [reply src/util/timeConverterToMinSec.ts:1]: And let’s call the function “millisToStr”

File: src/util/timeConverterToMinSec.test.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {timeConverterToMinSec} from "./timeConverterToMinSec";
+describe('TimeConverter', () => {
+    it('Converts time correctly', () => {
+
+        expect(timeConverterToMinSec(1000)).toEqual(("0:01"));
+        expect(timeConverterToMinSec(2050)).toEqual(("0:02"));
+        expect(timeConverterToMinSec(30000)).toEqual(("0:30"));
```
Comment (vck3000) [inline src/util/timeConverterToMinSec.test.ts:7]: These should be 00:02 etc.

Comment (vck3000) [reply src/util/timeConverterToMinSec.test.ts:7]: Actually, I could be convinced otherwise. Maybe ok to keep as is. I’ll leave it up to you.

Comment (kev306) [reply src/util/timeConverterToMinSec.test.ts:7]: Decided to update :)

File: src/util/time.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+export function millisToStr(millis: number) : string {
+
```
Comment (vck3000) [inline src/util/time.ts:2]: Nit: get rid of this new line

File: src/util/time.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+export function millisToStr(millis: number) : string {
+
+    let minutes: number = Math.floor(millis/ (1000 * 60) );
+    let seconds: number = Math.floor((millis / 1000) % 60);
+    let formattedSeconds: string;
```
Comment (vck3000) [inline src/util/time.ts:5]: Nit: New line above this

File: src/gameplay/game.ts [hunk 8]
```diff
@@ -21,6 +21,7 @@ import { avalonCards } from './cards/cards';
 import { avalonPhases, commonPhases } from './phases/phases';
 import { Card } from './cards/types';
 import { Role } from './roles/types';
+import {millisToStr} from "../util/time";
```
Comment (vck3000) [inline src/gameplay/game.ts:24]: Something is up with your prettier lol. Should have some spaces in the curly brackets.

File: src/gameplay/game.ts [hunk 9]
```diff
@@ -483,9 +484,9 @@ class Game extends Room {
     const timeouts = this.gameTimer.getTimeouts();
     if (timeouts.assassination || timeouts.default) {
       this.sendText(
-        `Timeouts: Default = ${timeouts.default / 1000}s Assassination = ${
-          timeouts.assassination / 1000
-        }s`,
+        `Timeouts: Default = ${millisToStr(timeouts.default)}, Assassination = ${
+            millisToStr(timeouts.assassination)
+        }`,
```
Comment (vck3000) [inline src/gameplay/game.ts:489]: Nice, very nice!

File: src/gameplay/room.ts [hunk 10]
```diff
@@ -644,10 +645,8 @@ class Room {
 
     rolesInStr += `<br>Ranked: ${this.ranked}`;
     rolesInStr += `<br>Mute Spectators: ${this.muteSpectators}`;
-    rolesInStr += `<br>Default timeout: ${timeouts.default / 1000}s`;
-    rolesInStr += `<br>Assassination timeout: ${
-      timeouts.assassination / 1000
-    }s`;
+    rolesInStr += `<br>Default timeout: ${millisToStr(timeouts.default)}`;
+    rolesInStr += `<br>Assassination timeout: ${millisToStr(timeouts.assassination)}`;
```
Comment (vck3000) [inline src/gameplay/room.ts:649]: Nice!!

File: src/util/time.test.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import {millisToStr} from "./time";
+describe('TimeConverter', () => {
+    it('Converts time correctly', () => {
+
+        //Test seconds displayed correctly
```
Comment (vck3000) [inline src/util/time.test.ts:5]: Nit: Should have space after //

File: src/util/time.test.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import {millisToStr} from "./time";
+describe('TimeConverter', () => {
```
Comment (vck3000) [inline src/util/time.test.ts:2]: Nit: New line above

File: src/util/time.test.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+        //Test minutes displayed correctly
+        expect(millisToStr(60000)).toEqual("01:00");
+        expect(millisToStr(660000)).toEqual("11:00");
+        expect(millisToStr(61005)).toEqual("01:01");
+
+        //Test min + sec displayed correctly
+        expect(millisToStr(62000)).toEqual("01:02");
+    })
+});
```
Comment (vck3000) [inline src/util/time.test.ts:18]: Should have new line at the end of this file. Something up with your editor.

**Comment by vck3000** [comment]
image

image

image

**Comment by vck3000** [comment]
Closes #593

File: src/gameplay/room.ts [hunk 14]
```diff
@@ -9,6 +9,7 @@ import { avalonCards } from './cards/cards';
 import { avalonPhases, commonPhases } from './phases/phases';
 import { Role } from './roles/types';
 import { Phase } from './phases/types';
+import {millisToStr} from "../util/time";
```
Comment (vck3000) [inline src/gameplay/room.ts:12]: Spaces

**Review by vck3000 — APPROVED** [review]

**Pull request merged**